R/method-correlation.R

```diff
@@ -0,0 +1,100 @@
+#' Correlation
+#'
+#' This function returns correlation coefficients for all abundance variables vs all continuous sample metadata variables.
+#' Soon this function will be expanded to take multiple abundance datasets, and to run all-to-all correlations
+#' for abundance variables.
+#' 
+#' @param data AbundanceData object
+#' @param method string defining the type of correlation to run. The currently supported values are 'spearman' and 'pearson'
+#' @param verbose boolean indicating if timed logging is desired
+#' @return ComputeResult object
+#' @import veupathUtils
+#' @import data.table
+#' @useDynLib microbiomeComputations
+#' @export
+setGeneric("correlation",
+  function(data, method = c('spearman','pearson'), verbose = c(TRUE, FALSE)) standardGeneric("correlation"),
+  signature = c("data")
+)
+
+#'@export
+setMethod("correlation", signature("AbundanceData"), function(data, method = c('spearman','pearson'), verbose = c(TRUE, FALSE)) {
+    df <- data@data
+    recordIdColumn <- data@recordIdColumn
+    naToZero <- data@imputeZero
+    ancestorIdColumns <- data@ancestorIdColumns
+    allIdColumns <- c(recordIdColumn, ancestorIdColumns)
+    sampleMetadata <- data@sampleMetadata
+
+    ## Initialize and check inputs
+    method <- veupathUtils::matchArg(method)
+    verbose <- veupathUtils::matchArg(verbose)
+
+    # Check that incoming df meets requirements
+    if (!'data.table' %in% class(df)) {
+      data.table::setDT(df)
+    }
+
+    # Check that incoming metadata meets requirements
+    if (!'data.table' %in% class(sampleMetadata)) {
+      data.table::setDT(sampleMetadata)
+    }
```

Comment:
are these not validation the relevant classes can perform? like can AbundanceData not enforce that data and sampleMetadata are of class data.table?

Reply:
they can enforce that class by erring, but they cant' change the class if it's not correct. So yeah we could totally just move all this validation to the class! That'd clean up all the methods i think.

Reply:
#46